src/OptionSettlementEngine.sol

```diff
@@ -283,77 +283,67 @@ contract OptionSettlementEngine is ERC1155, IOptionSettlementEngine {
     }
 
     /// @inheritdoc IOptionSettlementEngine
-    /// @dev Supplying claimId as 0 to the overloaded write signifies that a new
-    /// claim NFT should be minted for the options lot, rather than being added
-    /// as an existing claim.
-    function write(uint256 optionId, uint112 amount) external returns (uint256 claimId) {
-        return write(optionId, amount, 0);
-    }
-
-    /// @inheritdoc IOptionSettlementEngine
-    function write(uint256 optionId, uint112 amount, uint256 claimId) public returns (uint256) {
-        (uint160 optionKey, uint96 decodedClaimNum) = decodeTokenId(optionId);
-
-        // optionId must be zero in lower 96b for provided option Id
-        if (decodedClaimNum != 0) {
-            revert InvalidOption(optionId);
-        }
-
-        // claim provided must match the option provided
-        if (claimId != 0 && ((claimId >> 96) != (optionId >> 96))) {
-            revert EncodedOptionIdInClaimIdDoesNotMatchProvidedOptionId(claimId, optionId);
-        }
-
+    function write(uint256 tokenId, uint112 amount) public returns (uint256) {
+        // You need to write some amount
         if (amount == 0) {
             revert AmountWrittenCannotBeZero();
         }
 
+        (uint160 optionKey, uint96 claimNum) = decodeTokenId(tokenId);
+        uint256 encodedClaim = tokenId;
+        uint256 encodedOption = encodeTokenId(optionKey, 0);
+
+        // Get the option record and check that it's valid to write against
         Option storage optionRecord = _option[optionKey];
 
         uint40 expiry = optionRecord.expiryTimestamp;
         if (expiry == 0) {
             revert InvalidOption(optionKey);
         }
         if (expiry <= block.timestamp) {
-            revert ExpiredOption(optionId, expiry);
+            revert ExpiredOption(encodedOption, expiry);
         }
 
+        // Calculate the amount of underlying and exercise assets to transfer
         uint256 rxAmount = amount * optionRecord.underlyingAmount;
-        uint256 fee = ((rxAmount / 10_000) * feeBps);
+        uint256 fee = ((rxAmount * feeBps) / 10_000);
```

Comment:
in the future, we should probably contain PRs to only address what was mentioned in the audit to minimize the delta they need to review